common/src/main/java/com/netflix/conductor/common/run/Workflow.java

```diff
@@ -444,9 +383,8 @@ public String getWorkflowName() {
      * @return the workflow definition version.
      */
     public int getWorkflowVersion() {
-        return getWorkflowDefinition() != null ?
-            getWorkflowDefinition().getVersion() :
-            version;
+        Preconditions.checkNotNull(workflowDefinition, "Workflow definition is null");
+        return getWorkflowDefinition().getVersion();
```

Comment:
is this indirection needed? Similar to the method above this could be `workflowDefinition.getVersion()`